sql/catalog/lease: TestSubqueryLeases failed

sql/catalog/lease.TestSubqueryLeases [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21055944?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21055944?buildTab=artifacts#/) on master @ [2b6020f49881f6db70f89991556f798cbb4d44fb](https://github.com/cockroachdb/cockroach/commits/2b6020f49881f6db70f89991556f798cbb4d44fb):


```
=== RUN   TestSubqueryLeases
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestSubqueryLeases2964253070
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:183: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:876: CREATE TABLE has acquired a lease: got 1, expected 0
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestSubqueryLeases2964253070
--- FAIL: TestSubqueryLeases (2.64s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSubqueryLeases.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59343

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -839,6 +839,7 @@ func TestSubqueryLeases(t *testing.T) {
 	fooReleaseCount := int32(0)
 	var tableID int64
 	var params base.TestServerArgs
+	params.Settings = cluster.MakeTestingClusterSettings()
 	params.Knobs = base.TestingKnobs{
 		SQLLeaseManager: &lease.ManagerTestingKnobs{
 			LeaseStoreTestingKnobs: lease.StorageTestingKnobs{
@@ -861,6 +862,10 @@ func TestSubqueryLeases(t *testing.T) {
 			},
 		},
 	}
+	// Ensure wait for initial version is disabled, so that newly created
+	// descriptors are not automatically leased.
+	ctx := context.Background()
+	lease.WaitForInitialVersion.Override(ctx, &params.Settings.SV, false)
 	srv, sqlDB, kvDB := serverutils.StartServer(t, params)
 	defer srv.Stopper().Stop(context.Background())
 	s := srv.ApplicationLayer()

